### DISCUSSION TITLE
Using 3rd-party OpenAI-compatible providers

### DISCUSSION THREAD

**Discussion by johnd0e** [darrenburns/elia#35] [Ideas]
At the moment it is possible setting OPENAI_API_KEY / OPENAI_API_BASE variables.

But such setup does not allow to switch providers on-the-fly, or use some specific models.

Perhaps it could be implemented with some settings in `config.toml`.

My initial ideas:
- Each OPENAI_API_BASE value should be separate provider, with own name.
- The corresponding value of OPENAI_API_KEY should be stored in `config.toml` too.

**Comment by paw-lu** [comment]
i'm not getting elia to work well with base urls: #73 , something going wrong here?

**Comment by paw-lu** [comment]
> At the moment it is possible setting OPENAI_API_KEY / OPENAI_API_BASE variables.

like even if i set `OPENAI_API_KEY`, `OPENAI_API_BASE`, and `OPENAI_BASE_URL`, it doesn't seem to work out.
